src/hotspot/share/runtime/javaThread.cpp

```diff
@@ -1000,7 +1000,8 @@ bool JavaThread::is_lock_owned(address adr) const {
   assert(LockingMode != LM_LIGHTWEIGHT, "should not be called with new lightweight locking");
   if (Thread::is_lock_owned(adr)) return true;
 
-  for (MonitorChunk* chunk = monitor_chunks(); chunk != nullptr; chunk = chunk->next()) {
+  // Consider removing: chunk is always null.
+  for (MonitorChunk* chunk = monitor_chunks_safe(); chunk != nullptr; chunk = chunk->next()) {
```

Comment:
Unfortunately, this is now going to give the wrong answer if a deopt is in progress.  I think we need a better solution.